starknet_devnet/blocks.py

```diff
@@ -168,14 +176,15 @@ async def get_state_update(
         # now either an int or "latest"
         if block_number != LATEST_BLOCK_ID:
             self.__assert_block_number_in_range(block_number)
-            if block_number in self.__state_updates:
-                return self.__state_updates[block_number]
+            numeric_hash = self.__num2hash[block_number]
+            if numeric_hash in self.__state_updates:
+                return self.__state_updates[numeric_hash]
 
-            return await self.origin.get_state_update(block_number=block_number)
+            return await self.origin.get_state_update(block_hash=numeric_hash)
 
         # now we know the block ID is "latest"
         return (
-            self.__state_updates.get(self.get_number_of_blocks() - 1)
+            self.__state_updates[(await self.get_last_block()).block_hash]
```

Comment:
The part after `or` gets evaluated if the part before `or` is falsy (e.g. `None`). By using the `[key]` approach, we are risking `KeyError`, and by using the `.get(key)` approach, it really can be `None`.

Although if the comment "now we know the block ID is 'latest'" is true, then I'm not sure when this _defaulting when None_ is useful.

Reply:
Fixed in cf60a08603da192c7030c4479e383b187f717a80, I updated the comment - it was misleading since we are getting state from block_hash now, not block_number